Create and Deploy NPM Packages

cyberShaw opened the pull request:
### Article Checklist
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? see the example [Formats here](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)
- [x] Is your article properly formatted in Markdown? - see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)
- [x] Have you used the correct folder and file structure? - see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)
- [x] Is your article over 750 words?
- [x] Have you used our preferred heading structure? - [H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)
- [x] Does your article provide enough value and detail about your topic? Articles should be clear, accurate, and fully explained.
- [x] Can your article be understood by beginners? **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image? (600X400 pixels (images can be resized in paint) - also please make sure the image is from a site like www.unsplash.com)
- [x] In your PR - add a comment that states ["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278) to link and close your Idea Content (issue) when the PR is merged.
      

#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - Tool: [Grammarly](https://grammarly.com)
- [x] Have you checked your article for readability? - Tool: [Hemmingway](http://www.hemingwayapp.com/)
- [x] Have you added sources for quotes and images that aren't yours?

### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting/)
- [x] Have you used Repl.it for your code snippets?

### Contribution Guidelines
For first-time contributors and for more details, see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)

This fixes #349

louisefindlay23 commented on articles/npm-packages/index.md:
```diff
@@ -0,0 +1,191 @@
+    alt: NPM
+---
+
+Node.js is a popular server-side language that is used by millions of developers worldwide. While building Node.js projects, we use various packages that make development easier. All of these packages are managed by the Node Package Manager.
```
You could link to a Section article that explains what Node.js is. I see you've done so below but it would be great here as well.

louisefindlay23 commented on articles/npm-packages/index.md:
```diff
@@ -0,0 +1,191 @@
+
+### Introduction
+
+A package manager is a software that manages versions, packages and dependencies for your project. When a project grows in scale, a package manager helps to manage dependencies. It works perfectly across versions and even in collaborative projects. It cuts costs by reducing dependency problems across developer teams.
```
remove a in is a software.

louisefindlay23 commented on articles/npm-packages/index.md:
```diff
@@ -0,0 +1,191 @@
+
+A package manager is a software that manages versions, packages and dependencies for your project. When a project grows in scale, a package manager helps to manage dependencies. It works perfectly across versions and even in collaborative projects. It cuts costs by reducing dependency problems across developer teams.
+
+To install Node.js packages, we use the [NPM registry](https://www.npmjs.com/), which is a centralized repository of packages. In this article we will learn how to create a Node.js package and deploy it on the NPM registry for public use.
```
Add comma after In the article.

louisefindlay23 commented on articles/npm-packages/index.md:
```diff
@@ -0,0 +1,191 @@
+
+### Create an NPM Account
+
+To start up, let us create an NPM account.
```
let's would be more succinct.

louisefindlay23 commented on articles/npm-packages/index.md:
```diff
@@ -0,0 +1,191 @@
+
+![Create NPM Account](/engineering-education/npm-packages/npm-signup.png)
+
+- Now, you have to verify your account. Check your email ID that you provided and click on the verification email. This will verify your npm account.
```
Check the inbox of the email address you provided would make more sense and flow better.

louisefindlay23 commented on articles/npm-packages/index.md:
```diff
@@ -0,0 +1,191 @@
+
+### Let's Code
+
+- Every package published to the NPM registry should have a **`package.json`** file. Let us start by initializing a project.json file.
```
Let's would flow better. Consider revising further occurrences.

louisefindlay23 commented on articles/npm-packages/index.md:
````diff
@@ -0,0 +1,191 @@
+}
+```
+
+- In our sample NPM package, we are going to use the axios package for making API calls. Install it by using this command
````
You could link to the Axios package npm page here and/or provide further info about what the axios package does.

louisefindlay23 commented on articles/npm-packages/index.md:
```diff
@@ -0,0 +1,191 @@
+
+- Let us install and test our package.
+
+- Open a new folder and initialize a **`package.json`** by using the command,
```
Remove comma or replace it with a colon.

louisefindlay23 commented on articles/npm-packages/index.md:
````diff
@@ -0,0 +1,191 @@
+node index.js
+```
+
+- The contest data from our NPM package would be printed out on the console. We can do various other things using the retrieved data.
````
Perhaps provide a few suggestions on what readers could do with the retrieved data

louisefindlay23 reviewed (changes requested):
Great work, @cyberShaw. A well polished article. Keep the great quality up. I've made some small readability comments and suggested one or two areas you could expand on a little.

cyberShaw commented:
Thanks for the preliminary review, @louisefindlay23 . Those were some thoughtful changes that you suggested!

I've made the requested changes in the commit: da9ddd9

louisefindlay23 reviewed (approved):
Thanks, @cyberShaw for making the changes. Your article is now at the tech review stage.

louisefindlay23 commented on articles/npm-packages/index.md:
```diff
@@ -0,0 +1,194 @@
+
+### Let's Code
+
+- Every package published to the NPM registry should have a **`package.json`** file. Let's start by initializing a project.json file.
```
Should it not be initalizing a "**package.json**" file?

louisefindlay23 commented on articles/npm-packages/index.md:
```diff
@@ -0,0 +1,194 @@
+
+- Every package published to the NPM registry should have a **`package.json`** file. Let's start by initializing a project.json file.
+
+- The minimum requirements of a `package.json` file is a name and a version. Let's enter those information into the file.
```
those **pieces of** information would flow better.

louisefindlay23 commented on articles/npm-packages/index.md:
````diff
@@ -0,0 +1,194 @@
+  return result; // The contest data is returned
+};
+```
+
````
You should note to the readers that this code requires an API key and is not expected to run straight off the bat because you're just publishing the package.

louisefindlay23 commented on articles/npm-packages/index.md:
````diff
@@ -0,0 +1,194 @@
+- To deploy the package, enter the command:
+
+```bash
+npm publish
````
You need to instruct users to change the name of their package in `package.json` to a unique name otherwise they get an error.

louisefindlay23 commented on articles/npm-packages/index.md:
```diff
@@ -0,0 +1,194 @@
+
+### Further Reading
+
+For more knowledge on NPM Packages and JavaScript, check out the below articles.
```
lowercase p in packages.

louisefindlay23 reviewed (changes requested):
@cybershaw, great article. I tested your code and there are some points you need to make readers aware of otherwise they may encounter error. Add them in and your article should be ready for final review.

cyberShaw commented:
> @cyberShaw, great article. I tested your code and there are some points you need to make readers aware of otherwise they may encounter error. Add them in and your article should be ready for final review.

Thank you for the tech review! I have made the requested changes in commit: 0f74759

louisefindlay23 commented:
Great work, @cyberShaw. This is article is ready for review, @hectorkambow.

louisefindlay23 reviewed (approved):
Great work, @cyberShaw. This is article is ready for review, @hectorkambow.

hectorkambow commented:
Great article @cyberShaw  - I think it would wrap up the article nicely if we added a short conclusion outlining what we just went over- after that should be ready to publish. 🚀

cyberShaw commented:
> Great article @cyberShaw - I think it would wrap up the article nicely if we added a short conclusion outlining what we just went over- after that should be ready to publish. 🚀

Great idea, @hectorkambow ! I've added a neat summary outlining the steps we went through. Added them in this commit: 
e483efb
